### PR TITLE
Add custom topological order for services

### LINKED ISSUE #128: Specify a custom order for non dependent components for topological sort on the DAG
Service order:

1. zookeeper
2. hadoop
3. ranger
4. hdfs
5. yarn

All other service after without custom order.

### PR DESCRIPTION
**Pull request by rpignolet**
Fix #128

### CODE REVIEW

File: tdp/core/dag.py [hunk 1]
```diff
@@ -23,6 +23,19 @@
 
 logger = logging.getLogger("tdp").getChild("dag")
 
+SERVICES_ORDER = {
+    "zookeeper": 1,
+    "hadoop": 2,
+    "ranger": 3,
+    "hdfs": 4,
+    "yarn": 5,
+    "hive": 6,
```
Comment (nschung) [inline tdp/core/dag.py:32]: As discussed, the explicitly numbered order of Hive, Hbase and Spark make sense that there are an dependency between them. It's better that they have the same number and are sorted by lexicographic order IMO.

**Review by nschung — APPROVED** [review]

File: tdp/core/dag.py [hunk 2]
```diff
@@ -23,6 +23,19 @@
 
 logger = logging.getLogger("tdp").getChild("dag")
 
+SERVICES_ORDER = {
```
Comment (rpignolet) [inline tdp/core/dag.py:26]: Rename it `SERVICES_PRIORITIES`, this order do not change the DAG order, so it is more about priorities.

Comment (rpignolet) [inline tdp/core/dag.py:26]: Rename it `SERVICES_PRIORITIES`, this order do not change the DAG order, so it is more about priorities.

File: tdp/core/dag.py (hunk 1 above)
Comment (rpignolet) [reply tdp/core/dag.py:32]: As discuss we will keep explicite priorities for each service with a different number.

**Comment by leopaul36** [comment]
> Here's the order from this implementation:
> 
> Result from command `tdp nodes`:
> 
> ```yaml
> - zookeeper_client_install
> - zookeeper_server_install
> - zookeeper_kerberos_install
> - zookeeper_client_config
> - zookeeper_install
> - zookeeper_server_config
> - zookeeper_config
> - zookeeper_server_start
> - zookeeper_server_init
> - zookeeper_start
> - zookeeper_init
> - hadoop_client_install
> - hadoop_client_config
> - hadoop_install
> - hadoop_config
> - hadoop_start
> - hadoop_init
> - ranger_admin_install
> - ranger_usersync_install
> - ranger_kerberos_install
> - ranger_ssl-tls_install
> - ranger_admin_config
> - ranger_admin_start
> - ranger_admin_init
> - ranger_install
> - ranger_usersync_config
> - ranger_config
> - ranger_usersync_start
> - ranger_start
> - ranger_init
> - hdfs_client_install
> - hdfs_datanode_install
> - hdfs_journalnode_install
> - hdfs_namenode_install
> - hdfs_kerberos_install
> - hdfs_ranger_install
> - hdfs_ssl-tls_install
> - hdfs_datanode_config
> - hdfs_install
> - hdfs_journalnode_config
> - hdfs_namenode_config
> - hdfs_client_config
> - hdfs_namenode_formatzk
> - hdfs_journalnode_start
> - hdfs_journalnode_init
> - hdfs_ranger_config
> - hdfs_config
> - hdfs_namenode_start
> - hdfs_namenode_init
> - hdfs_datanode_start
> - hdfs_datanode_init
> - hdfs_ranger_init
> - hdfs_client_init
> - hdfs_start
> - hdfs_init
> - yarn_apptimelineserver_install
> - yarn_client_install
> - yarn_hdfs_init
> - yarn_mapred_jobhistoryserver_install
> - yarn_nodemanager_install
> - yarn_ranger_install
> - yarn_resourcemanager_install
> - yarn_kerberos_install
> - yarn_ssl-tls_install
> - yarn_apptimelineserver_config
> - yarn_apptimelineserver_start
> - yarn_install
> - yarn_mapred_jobhistoryserver_config
> - yarn_mapred_jobhistoryserver_start
> - yarn_nodemanager_config
> - yarn_resourcemanager_config
> - yarn_client_config
> - yarn_ranger_config
> - yarn_config
> - yarn_ranger_init
> - yarn_resourcemanager_start
> - yarn_nodemanager_start
> - yarn_client_init
> - yarn_start
> - yarn_init
> - hive_client_init
> - hive_client_install
> - hive_hdfs_init
> - hive_hiveserver2_install
> - hive_metastore_install
> - hive_kerberos_install
> - hive_ranger_install
> - hive_ssl-tls_install
> - hive_client_config
> - hive_hiveserver2_config
> - hive_install
> - hive_metastore_config
> - hive_init_schema
> - hive_metastore_start
> - hive_hiveserver2_start
> - hive_hiveserver2_init
> - hive_ranger_config
> - hive_config
> - hive_ranger_init
> - hive_start
> - hive_init
> - hbase_client_install
> - hbase_hdfs_init
> - hbase_master_install
> - hbase_phoenix_client_install
> - hbase_phoenix_client_config
> - hbase_phoenix_queryserver_client_install
> - hbase_phoenix_queryserver_client_config
> - hbase_phoenix_queryserver_daemon_install
> - hbase_phoenix_kerberos_install
> - hbase_phoenix_ssl-tls_install
> - hbase_phoenix_queryserver_daemon_config
> - hbase_ranger_install
> - hbase_regionserver_install
> - hbase_phoenix_coprocessor_install
> - hbase_rest_install
> - hbase_kerberos_install
> - hbase_client_config
> - hbase_ssl-tls_install
> - hbase_install
> - hbase_master_config
> - hbase_master_start
> - hbase_regionserver_config
> - hbase_ranger_config
> - hbase_ranger_init
> - hbase_phoenix_ranger_init
> - hbase_regionserver_start
> - hbase_rest_config
> - hbase_config
> - hbase_rest_start
> - hbase_phoenix_queryserver_daemon_start
> - hbase_start
> - hbase_init
> - spark_client_install
> - spark_hdfs_init
> - spark_historyserver_init
> - spark_historyserver_install
> - spark_kerberos_install
> - spark_client_config
> - spark_ssl-tls_install
> - spark_historyserver_config
> - spark_historyserver_start
> - spark_install
> - spark_config
> - spark_start
> - spark_init
> - knox_gateway_install
> - knox_kerberos_install
> - knox_ranger_install
> - knox_ssl-tls_install
> - knox_gateway_config
> - knox_install
> - knox_ranger_config
> - knox_config
> - knox_gateway_start
> - knox_ranger_init
> - knox_start
> - knox_init
> ```

Looks good!

**Review by leopaul36 — APPROVED** [review]

**Pull request merged**